Mathlib/NumberTheory/LSeries/Convergence.lean

```diff
@@ -0,0 +1,96 @@
+/-
+Copyright (c) 2024 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+import Mathlib.NumberTheory.LSeries.Basic
+import Mathlib.Topology.Instances.EReal
+
+/-!
+# Convergence of L-series
+
+We define `ArithmeticFunction.abscissaOfAbsConv f` (as an `EReal`) to be the infimum
+of all real numbers `x` such that the L-series of `f` converges for complex arguments with
+real part `x` and provide some results about it.
+
+## Tags
+
+L-series, abscissa of convergence
+-/
+
+namespace ArithmeticFunction
+
+open Complex Nat
+
+/-- The abscissa `x : EReal` of absolute convergence of the L-series associated to `f`:
+the series converges absolutely at `s` when `re s > x` and does not converge absolutely
+when `re s < x`. -/
+noncomputable def abscissaOfAbsConv (f : ArithmeticFunction ℂ) : EReal :=
+  sInf <| Real.toEReal '' {x : ℝ | LSeriesSummable f x}
+
+/-- An open right half plane is open. -/
+lemma _root_.Complex.isOpen_rightHalfPlane (x : EReal) : IsOpen {z : ℂ | x < z.re} :=
+  isOpen_lt continuous_const <| EReal.continuous_coe_iff.mpr continuous_re
```

Comment:
I think this probably belongs elsewhere. There is a similar lemma (with `im` rather than `re`) inlined in `differentiableAt_jacobiTheta₂` (with a slightly different proof). It might be desirable to unify these somewhere, and treat left / right / upper / lower halfplanes together.

Reply:
I don't think it makes much sense to "unify" them in the sense that one has one statement covering all of them. But it certainly makes sense to have four such lemmas together in a suitable place. What would be a reasonable file for them to live in?

Reply:
Yes, that's what I meant (four separate lemmas in the same file). I don't have a view on which file this should be, but probably the `#find_home` routine will.

Reply:
`#find_home(!)` does not produce anything useful. I think it is best to add a new file `Mathlib.Analysis.Complex.HalfPlane` (which imports `Mathlib.Analysis.Complex.Basic` and `Mathlib.Topology.Instances.EReal`) for these four lemmas.

Reply:
(I do not actually use the fact that a right half-plane is open in this PR, but it will come in handy in later contributions.)